universe/Ship.h

```diff
@@ -1,11 +1,15 @@
 #ifndef _Ship_h_
 #define _Ship_h_
 
+#include "ShipDesign.h"
```

Comment:
I'd probably move `INVALID_DESIGN_ID` out of the `ShipDesign` class rather than require this include...